### PR TITLE
Fix torchaudio installation

### PR DESCRIPTION
**Pull request by artbataev**
Signed-off-by: Vladimir Bataev <vbataev@nvidia.com>

# What does this PR do ?

Fixes Docker builds with `torchaudio` library:
- Fixes installation in `pytorch:22.04-py3`-based and other old containers (silently failed, which resulted in no `torchaudio` at all in the container)
- The build will fail if `torchaudio` is not installed at all (no CPU version). Warning if the GPU version is not compiled.

**NB**: we will still need to update this script after PyTorch 2.0 release.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [x] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [x] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Comment by GNroy** [comment]
Thank you for this PR!
I suggest you add a warning for the case when the major torch version changes to 2, as this will most likely break the installation logic.

**Review by titu1994 — COMMENTED** [review]
Thanks for the fix in advance !

**Comment by artbataev** [comment]
> I suggest you add a warning for the case when the major torch version changes to 2, as this will most likely break the installation logic.

Done

**Review by GNroy — APPROVED** [review]
LGTM apart from my comment below.

File: scripts/installers/install_torchaudio_latest.sh [hunk 1]
```diff
@@ -26,22 +26,40 @@ LATEST_RELEASE=$(git -c 'versionsort.suffix=-' \
 # expected TORCHAUDIO_BUILD_VERSION=*.**.*
 TORCHAUDIO_BUILD_VERSION=${LATEST_RELEASE:8:1}${PYTORCH_VERSION:1:5}
 
+TORCH_MAJOR_VERSION=$(python3 -c "major_version = \"${PYTORCH_VERSION}\".split('.')[0]; print(major_version)")
+TORCH_MINOR_VERSION=$(python3 -c "minor_version = \"${PYTORCH_VERSION}\".split('.')[1]; print(minor_version)")
+TORCHAUDIO_MINOR_VERSION=$(python3 -c "minor_version = \"${LATEST_RELEASE}\".rsplit('.')[-1]; print(minor_version)")
+
+if [[ $TORCH_MAJOR_VERSION -ne 1 ]]; then
+    echo "WARNING: Pytorch major version different from 1 not supported"
```
[the remark this answers is not in the document]
Comment (titu1994) [reply scripts/installers/install_torchaudio_latest.sh:34]: Torch audio is not a mandatory library, I'd rather it be ignored than break an install.

**Review by titu1994 — APPROVED** [review]

**Comment by titu1994** [comment]
We'll need to have a note somewhere that forces us to update this for when PT 2 comes out

**Pull request merged**